x/stakeibc/keeper/reward_converter.go

```diff
@@ -145,24 +149,28 @@ func (k Keeper) CalculateRewardsSplitAfterRebate(
 	if !chainHasRebate {
 		strideFeeAmount = sdk.NewDecFromInt(rewardsAmount).Mul(totalFeeRate).TruncateInt()
 	} else {
-		// It shouldn't be possible to have 0 delegations (since there are rewards and there was a community pool stake)
+		// Get supply of stTokens to determine the portion of TVL that the community pool liquid stake makes up
+		stDenom := utils.StAssetDenomFromHostZoneDenom(hostZone.HostDenom)
```

Comment:
Do you think it's worth consolidating some of the code in this function and `CalculateRewardsSplitBeforeRebate` into a helper, to reduce code duplication? It could make things slightly easier to follow. These are relatively small functions though, so would understand if you'd rather keep as is

Reply:
yeah I took a pass it it in the other PR but the initial reaction from people was that it actually made the code a little harder to follow / it was nice having all the math in one place

I may take another stab at it, but I think I _slightly_ lean towards keeping it as it is for now

Reply:
That makes sense, would definitely believe it would make it harder to follow. Let's keep as is, resolving!